openshift-client/src/main/java/io/fabric8/openshift/client/OpenShiftConfig.java

```diff
@@ -96,7 +96,16 @@ private static String getDefaultOapiVersion(Config config) {
   }
 
   private static String getDefaultOpenShiftUrl(Config config) {
-    return Utils.getSystemPropertyOrEnvVar(OPENSHIFT_URL_SYTEM_PROPERTY, URLUtils.join(config.getMasterUrl(), "oapi", getDefaultOapiVersion(config)));
+    String openshiftUrl = Utils.getSystemPropertyOrEnvVar(OPENSHIFT_URL_SYTEM_PROPERTY);
+    if (openshiftUrl != null) {
+      // The OPENSHIFT_URL environment variable may be provided without the 'oapi' path in some configurations
+      if (!openshiftUrl.contains("/oapi/")) {
```

Comment:
Should this check should really be is the URL path empty or `/` only? Either way I think it will be mostly be fine I guess.